File: ios/Classes/TiNotificationbannerModule.m
/**
 * titanium-notification-banner
 *
 * Created by Your Name
 * Copyright (c) 2018 Your Company. All rights reserved.
 */

#import "TiNotificationbannerModule.h"
#import "TiBase.h"
#import "TiHost.h"
#import "TiUtils.h"

@implementation TiNotificationbannerModule

#pragma mark Internal

// This is generated for your module, please do not change it
- (id)moduleGUID
{
  return @"add9865d-fbad-468b-b044-ba9888544dcc";
}

- (NSString *)moduleId
{
  return @"ti.notificationbanner";
}

#pragma mark Lifecycle

- (void)startup
{
  [super startup];
  DebugLog(@"[DEBUG] %@ loaded", self);
}

#pragma Public APIs

- (void)show:(id)args
{
  ENSURE_SINGLE_ARG(args, NSDictionary);

  NSString *title = args[@"title"];
  NSString *subtitle = args[@"subtitle"];
  double duration = [TiUtils doubleValue:@"duration" properties:args def:1.0];
  UIImage *image = [TiUtils image:args[@"image"] proxy:self];
  UIColor *backgroundColor = [TiUtils colorValue:@"backgroundColor" properties:args].color;
  Banner *banner = [[Banner alloc] initWithTitle:title
                                        subtitle:subtitle
                                           image:image
                                 backgroundColor:backgroundColor
                                     didTapBlock:^{ /* Unused */ }];
  banner.dismissesOnTap = YES;
  [banner show:nil duration:duration];
}

@end

File: ios/Classes/TiNotificationbannerModule.swift
//
//  TiNotificationbannerModule.swift
//  titanium-notification-banner
//
//  Created by Hans Knöchel
//  Copyright (c) 2019-present by Lambus GmbH. All rights reserved.
//

import BRYXBanner
import UIKit
import TitaniumKit

@objc(TiNotificationbannerModule)
class TiNotificationbannerModule: TiModule {

  public let testProperty: String = "Hello World"
  
  func moduleGUID() -> String {
    return "cccc9652-8055-43e0-a933-5cc76acd00be"
  }
  
  override func moduleId() -> String! {
    return "ti.notificationbanner"
  }

  override func startup() {
    super.startup()
    debugPrint("[DEBUG] \(self) loaded")
  }

  @objc(show:)
  func show(arguments: Array<Any>?) {
    guard let arguments = arguments, let params = arguments[0] as? [String: Any] else { return }
   
    let title = params["title"] as? String
    let subtitle = params["subtitle"] as? String
    let duration = params["duration"] as? Double
    let image = params["image"] as? String
    let titleColor = params["titleColor"] as? String
    let backgroundColor = params["backgroundColor"] as? String
    let onClickCallback = params["onClick"] as? KrollCallback
    
    let banner = Banner(title: title, subtitle: subtitle, image: TiUtils.toImage(image, proxy: self), backgroundColor: TiUtils.colorValue(backgroundColor)?.color ?? UIColor.clear) {
      if let onClickCallback = onClickCallback {
        onClickCallback.call([[:]], thisObject: self)
      }
    }
    
    if let titleColor = TiUtils.colorValue(titleColor) {
      banner.textColor = titleColor.color
    }

    banner.dismissesOnTap = true
    banner.dismissesOnSwipe = true

    banner.show(duration: duration)
  }
}
